### PR TITLE
Implemented ttl. Resolves #4.

### LINKED ISSUE #4: Add timed cached_property
Must work like this:

``` python
class Monopoly(object):

    def __init__(self):
        self.boardwalk_price = 500
        self.parkplace_price = 450

    # No time specified so cache is indefinite
    @timed_cached_property
    def boardwalk(self):
        self.boardwalk_price += 50
        return self.boardwalk_price

    # Time of 300 specified, making cache last for 5 minutes before invalidation
    @timed_cached_property(ttl=300)
    def parkplace(self):
        self.timed_cached_property += 50
        return self.timed_cached_property
```

Notes:

* This *might* supplant the current `cached_property` decorator.
* The example at https://wiki.python.org/moin/PythonDecoratorLibrary#Cached_Properties breaks if you don't pass a TTL (time-to-leave) argument.

### PR DESCRIPTION
**Pull request by thedrow**

### CODE REVIEW

**Comment by pydanny** [comment]
Oh no! I missed this PR! Want to work on it still?

**Comment by thedrow** [comment]
Yes but I have no idea why the tests are failing.

**Comment by pydanny** [comment]
I worked on it for about 30 minutes. Have to get to work now, but the relationship between `__call__` and `__get__` is tricky. Fun fun!

File: cached_property.py [hunk 1]
```diff
@@ -16,23 +17,44 @@ class cached_property(object):
...
-        value = obj.__dict__[self.func.__name__] = self.func(obj)
-        return value
+        if not self.ttl:
+            value = obj.__dict__[self.func.__name__] = self.func(obj)
+            return value
+        else:
+            now = time.time()
+            key = '_cached_' + self.func.__name__
+            try:
+                value, last_update = obj.__dict__[key]
```
Comment (bcho) [inline cached_property.py]: Instead of using `__dict__`, you may have to use another dunder field (like `_cache`).

**Comment by thedrow** [comment]
Superceeded by #14.

**Pull request closed without merging**